src/world/terrain/material.hpp

```diff
@@ -49,22 +53,73 @@ namespace terrain {
 struct Material {
     Material(
         std::vector<std::pair<const std::string, ColorInt>> color_in,
-        float speed_multiplier_in, bool solid_in, MaterialId element_id_in,
+        float speed_multiplier_in, bool solid_in, MaterialId material_id_in,
         std::string name_in
     ) :
         color(color_in),
         speed_multiplier(speed_multiplier_in), solid(solid_in),
-        element_id(element_id_in), name(name_in){};
+        material_id(material_id_in), name(name_in){};
     // vector of <name hex color> for possible colors
     std::vector<std::pair<const std::string, ColorInt>> color;
-    float speed_multiplier = 1;         // speed on this material
-    bool solid = false;                 // Is the material solid?
-    MaterialId element_id = AIR_MAT_ID; // The ID of the material (Air is 0)
-    const std::string name = "Air";     // The material name
+    float speed_multiplier = 1;          // speed on this material
+    bool solid = false;                  // Is the material solid?
+    MaterialId material_id = AIR_MAT_ID; // The ID of the material (Air is 0)
+    const std::string name = "Air";      // The material name
     // int8_t deterioration from wind
     // int8_t deterioration from water
 };
 
+struct material_color_t {
+    std::string color_name;
+    ColorInt hex_color;
+
+    void
+    read_hex_color(std::string hex_string) {
+        hex_color = std::stoll(hex_string, 0, 16);
```

Comment:
```suggestion
        hex_color = std::stoll(hex_string, nullptr, 16);
```

ALEM STOP USING 0 FOR nullptr